foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/descriptors/ClassDescriptor.java

```diff
@@ -3993,11 +4054,63 @@ public void postInitialize(AbstractSession session) throws DescriptorException {
 
         getCachePolicy().postInitialize(this, session);
 
+        postInitializeReturningPolicies();
+
         validateAfterInitialization(session);
 
         checkDatabase(session);
     }
 
+    private void postInitializeReturningPolicies() {
+        //Initialize ReturningPolicies
+        List<ReturningPolicy> returningPolicies = new ArrayList<>();
+        if (this.hasReturningPolicy()) {
+            returningPolicies.add(this.getReturningPolicy());
+        }
+        for (DatabaseMapping databaseMapping :this.getMappings()) {
+            if (databaseMapping instanceof AggregateObjectMapping) {
+                ClassDescriptor referenceDescriptor = databaseMapping.getReferenceDescriptor();
```

Comment:
EclipseLink does support nested aggregates so you will need to check the mappings of the reference descriptor too.  If the AggregateObject's ClassDescriptor has already been initialized you may be able to just get it's list of return policies.

Reply:
It's changed into recursive method to analyze multi-level embedded entities/objects.